http-utils/src/main/java/ai/vespa/util/http/hc5/VespaHttpClientBuilder.java

```diff
@@ -10,47 +11,75 @@
 import org.apache.hc.client5.http.ssl.SSLConnectionSocketFactory;
 import org.apache.hc.core5.http.config.Registry;
 import org.apache.hc.core5.http.config.RegistryBuilder;
+import org.apache.hc.core5.util.Timeout;
 
 import javax.net.ssl.HostnameVerifier;
 import javax.net.ssl.SSLParameters;
 
+import java.util.concurrent.TimeUnit;
+
 import static com.yahoo.security.tls.MixedMode.PLAINTEXT_CLIENT_MIXED_SERVER;
 import static com.yahoo.security.tls.TransportSecurityUtils.getInsecureMixedMode;
 import static com.yahoo.security.tls.TransportSecurityUtils.getSystemTlsContext;
 import static com.yahoo.security.tls.TransportSecurityUtils.isTransportSecurityEnabled;
 
 /**
  * Sync HTTP client builder <em>for internal Vespa communications over http/https.</em>
- *
  * Configures Vespa mTLS and handles TLS mixed mode automatically.
- * Custom connection managers must be configured through {@link #create(HttpClientConnectionManagerFactory)}.
+ * Custom connection managers must be configured through {@link #set(HttpClientConnectionManagerFactory)}.
  *
  * @author jonmv
  */
 public class VespaHttpClientBuilder {
+    private HttpClientConnectionManagerFactory connectionManagerFactory = PoolingHttpClientConnectionManager::new;
+    private HostnameVerifier hostnameVerifier = new NoopHostnameVerifier();
+    private boolean rewriteHttpToHttps = true;
+    private final ConnectionConfig.Builder connectionConfigBuilder = ConnectionConfig.custom();
 
     public interface HttpClientConnectionManagerFactory {
-        HttpClientConnectionManager create(Registry<ConnectionSocketFactory> socketFactories);
+        PoolingHttpClientConnectionManager create(Registry<ConnectionSocketFactory> socketFactories);
     }
 
-    public static HttpClientBuilder create() {
-        return create(PoolingHttpClientConnectionManager::new);
+    private VespaHttpClientBuilder() {
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory) {
-        return create(connectionManagerFactory, new NoopHostnameVerifier());
+    public static VespaHttpClientBuilder custom() {
+        return new VespaHttpClientBuilder();
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier) {
-        return create(connectionManagerFactory, hostnameVerifier, true);
+    public VespaHttpClientBuilder set(HttpClientConnectionManagerFactory connectionManagerFactory) {
+        this.connectionManagerFactory = connectionManagerFactory;
+        return this;
     }
 
-    public static HttpClientBuilder create(HttpClientConnectionManagerFactory connectionManagerFactory,
-                                           HostnameVerifier hostnameVerifier,
-                                           boolean rewriteHttpToHttps) {
+    public VespaHttpClientBuilder set(HostnameVerifier hostnameVerifier) {
```

Comment:
Consider renaming to `hostnameVerifier()`

Reply:
Fixed